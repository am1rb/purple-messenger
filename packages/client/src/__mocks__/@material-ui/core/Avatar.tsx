import React from "react";
import type { AvatarProps } from "@material-ui/core/Avatar";

function MuiAvatar({ src, alt }: AvatarProps) {
  return <img data-testid="mock-mui-avatar" src={src} alt={alt} />;
}

export default MuiAvatar;
