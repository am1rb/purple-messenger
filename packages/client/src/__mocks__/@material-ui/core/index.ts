export { default as Avatar } from "./Avatar";
export { default as Box } from "./Box";
export { default as Card } from "./Card";
export { default as CardContent } from "./CardContent";
export { default as Typography } from "./Typography";
export { default as Button } from "./Button";
