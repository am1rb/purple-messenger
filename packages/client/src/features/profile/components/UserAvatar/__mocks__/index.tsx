import React from 'react';

const userAvatar = jest.fn(() => <div data-testid="user-avatar" />);

export default userAvatar;
