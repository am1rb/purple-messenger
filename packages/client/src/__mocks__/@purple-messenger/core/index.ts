import * as purpleMessengerCore from "@purple-messenger/core";

const mockedPurpleMessengerCore = jest.genMockFromModule<
  typeof purpleMessengerCore
>("@purple-messenger/core");

function mockedAction(action = () => ({ type: "" })) {
  return jest.fn(action);
}

module.exports = {
  ...mockedPurpleMessengerCore,
  loadConversationList: mockedAction(),
  unloadConversationList: mockedAction(),
  signOut: mockedAction(),
  clearMessageList: mockedAction(),
  seenMessageAck: mockedAction(),
  start: mockedAction(),
  shutdown: mockedAction(),
  submitMessage: mockedAction(),
  startTypingMessage: mockedAction(),
  stopTypingMessage: mockedAction(),
  signIn: mockedAction(),
  setAuthError: mockedAction(),
  verifyToken: mockedAction(),
  connected: mockedAction(purpleMessengerCore.connected),
  disconnected: mockedAction(purpleMessengerCore.disconnected),
};
