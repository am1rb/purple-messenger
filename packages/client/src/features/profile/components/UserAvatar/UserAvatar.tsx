import React, { memo } from "react";
import { useSelector } from "react-redux";
import clsx from "clsx";
import Avatar from "@material-ui/core/Avatar";
import { getProfileInfoImage } from "features/profile/selectors";
import useStyles from "./UserAvatar.styles";

interface Props {
  className?: string;
}

function UserAvatar({ className }: Props) {
  const classes = useStyles();
  const profileInfoImage = useSelector(getProfileInfoImage);

  return (
    <Avatar className={clsx(classes.root, className)} src={profileInfoImage} />
  );
}

export default memo(UserAvatar);
