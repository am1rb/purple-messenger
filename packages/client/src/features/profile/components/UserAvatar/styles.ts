import { makeStyles } from "@material-ui/core";

export default makeStyles({
  root: {
    width: 80,
    height: 80,
    margin: "0 auto"
  }
});
