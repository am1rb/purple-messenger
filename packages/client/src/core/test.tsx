import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { ConnectedRouter } from "connected-react-router";
import configureStore, { history } from "core/redux/configureStore";
import { State as RootState } from "core/redux/reducers";
import theme from "core/style/theme";
import { ThemeProvider } from "@material-ui/styles";

export function renderWithStore(
  ui: React.ReactElement,
  initStore: Partial<RootState>
) {
  const store = configureStore(initStore);

  return render(ui, {
    wrapper: ({ children }) => (
      <ThemeProvider theme={theme}>
        <Provider store={store}>
          <ConnectedRouter history={history}>{children}</ConnectedRouter>
        </Provider>
      </ThemeProvider>
    ),
  });
}
