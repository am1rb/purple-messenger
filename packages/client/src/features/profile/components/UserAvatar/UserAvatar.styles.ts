import { Theme } from "@material-ui/core";
import { makeStyles } from "@material-ui/styles";

export default makeStyles(
  (theme: Theme) => ({
    root: {
      width: theme.spacing(10),
      height: theme.spacing(10),
      margin: "0 auto",
    },
  }),
  { name: "UserAvatar" }
);
