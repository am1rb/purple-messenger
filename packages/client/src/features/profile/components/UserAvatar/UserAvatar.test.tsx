import React from "react";
import { sampleProfileInfo } from "@purple-messenger/core";
import { renderWithStore } from "core/test";
import UserAvatar from "./UserAvatar";

describe("The <UserAvatar /> tests", () => {
  it("Should match the snapshot", () => {
    const { container } = renderWithStore(<UserAvatar />, {
      profile: sampleProfileInfo,
    });
    expect(container).toMatchSnapshot();
  });
});
