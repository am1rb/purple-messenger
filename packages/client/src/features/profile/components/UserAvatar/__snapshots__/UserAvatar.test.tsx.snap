// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The <UserAvatar /> tests Should match the snapshot 1`] = `
<div>
  <img
    data-testid="mui-avatar"
    src="https://picsum.photos/200"
  />
</div>
`;
